code/datums/martial/sleeping_carp.dm

```diff
@@ -98,42 +116,44 @@
 	log_combat(A, D, "disarmed (Sleeping Carp)")
 	return ..()
 
-/datum/martial_art/the_sleeping_carp/on_projectile_hit(mob/living/A, obj/projectile/P, def_zone)
-	. = ..()
-	if(A.incapacitated(IGNORE_GRAB)) //NO STUN
-		return BULLET_ACT_HIT
-	if(!(A.mobility_flags & MOBILITY_USE)) //NO UNABLE TO USE
-		return BULLET_ACT_HIT
-	var/datum/dna/dna = A.has_dna()
+/datum/martial_art/the_sleeping_carp/proc/can_deflect(mob/living/carp_user)
+	if(carp_user.incapacitated(IGNORE_GRAB)) //NO STUN
+		return FALSE
+	if(!(carp_user.mobility_flags & MOBILITY_USE)) //NO UNABLE TO USE
+		return FALSE
+	var/datum/dna/dna = carp_user.has_dna()
 	if(dna?.check_mutation(/datum/mutation/human/hulk)) //NO HULK
 		return BULLET_ACT_HIT
-	if(!isturf(A.loc)) //NO MOTHERFLIPPIN MECHS!
+	if(!isturf(carp_user.loc)) //NO MOTHERFLIPPIN MECHS!
+		return BULLET_ACT_HIT
+	return TRUE
```

Comment:
the return value is too inconsistent here, you're switching between defines and booleans. Stick to one

Reply:
ahh, true